backend/routes/youchoose.js

```diff
@@ -58,15 +58,17 @@ async function byVideoId(req) {
     ? await ycai.fetchChannelRecommendations(channelId)
     : [];
 
+  const fromChannel = channelRecommendations
+    .filter(({ urlId }) => !avail.find(({ urlId: id }) => id === urlId))
+    .map((rec) => ({
+      ...rec,
+      fromChannel: true,
+    }));
+
+  fromChannel.sort(() => Math.random() - 0.5);
```

Comment:
by doing this we always have a sorted list of the first 20 recommendations, isn't this implementable at the db query level?

Reply:
The real randomization happens in the `$sample` stage of the aggregate pipeline.
I'm only adding this hackish sort because if your db has < 20 recommendations mongo will always return them in the same order.